CesiumGltfContent/include/CesiumGltfContent/GltfUtilities.h

```diff
@@ -191,5 +195,44 @@ struct CESIUMGLTFCONTENT_API GltfUtilities {
    * @param bufferIndex The index of the buffer to compact.
    */
   static void compactBuffer(CesiumGltf::Model& gltf, int32_t bufferIndex);
+
+  /**
+   * @brief Hit result data for intersectRayGltfModelParametric
+   */
+  struct IntersectResult {
+    struct Hit {
+      glm::dvec3 primitivePoint = {};
+      glm::dmat4x4 primitiveToWorld = {};
+
+      glm::dvec3 worldPoint = {};
+      double rayToWorldPointDistanceSq = -1;
+
+      int meshId = -1;
+      int primitiveId = -1;
+    };
+
+    std::optional<Hit> hit;
+
+    std::vector<std::string> warnings{};
+  };
+
+  /**
+   * @brief Intersects a ray with a glTF model and returns the first
+   * intersection point.
+   *
+   * Supports all mesh primitive modes.
+   * Points and lines are assumed to have no area, and are ignored
+   *
+   * @param ray A ray in world space.
+   * @param gltf The glTF model to intersect.
+   * @param cullBackFaces Ignore triangles that face away from ray.
```

Comment:
Perhaps the default should be to do what the primitive's material says to do?
https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#double-sided

In that case you'd need an enumeration instead of a bool if you also want to let the user override it.